back/engines/commercial/geographic_dashboard/spec/acceptance/geotag_ideas_spec.rb

```diff
@@ -20,27 +20,52 @@
   end
 
   get '/web_api/v1/ideas/geotagged' do
-    let!(:ideas) do
-      create_list(:idea, 2, title_multiloc: { 'nl-BE' => 'Een fietspad in het Prinsenbos van Grimbergen' })
-    end
-
-    before do
-      allow_any_instance_of(GeographicDashboard::GeotagService).to receive(:geotag).and_return(geocoded)
-      do_request
-    end
-
     with_options scope: :page do
       parameter :number, 'Page number'
       parameter :size, 'Number of campaigns per page'
     end
 
-    example 'Returns the right status code' do
-      expect(status).to eq 200
+    context "When ideas have same locale" do
+      let!(:ideas) do
+        create_list(:idea, 2, title_multiloc: { 'nl-BE' => 'Een fietspad in het Prinsenbos van Grimbergen' })
+      end
+
+      before do
+        allow_any_instance_of(NLP::Api).to receive(:geotag).with(
+          instance_of(String), instance_of(String), instance_of(String), anything
+        ).and_return([geocoded])
+        do_request
```

Comment:
`do_request` is typically part of the test, either explicitly or as part of an `example_request`.